src/webui/api/appcontroller.cpp

```diff
@@ -1083,6 +1085,24 @@ void AppController::defaultSavePathAction()
     setResult(BitTorrent::Session::instance()->savePath().toString());
 }
 
+void AppController::getDirectoriesAction()
+{
+    requireParams({u"path"_s});
+
+    const QString path = params().value(u"path"_s);
+    if (path.isEmpty() || path.startsWith(u':'))
+        throw APIError(APIErrorType::BadParams);
+
+    const QDir dir {path};
+    if (!dir.isAbsolute())
+        throw APIError(APIErrorType::BadParams);
+    if (!dir.exists())
+        throw APIError(APIErrorType::NotFound);
+
+    const QStringList dirs = dir.entryList(QDir::Dirs | QDir::NoDotAndDotDot);
+    setResult(QJsonArray::fromStringList(dirs));
```

Comment:
I would avoid redundant temporary string list by using:
```suggestion
    QJsonArray ret;
    QDirIterator it {path, (QDir::Dirs | QDir::NoDotAndDotDot)};
    while (it.hasNext())
    {
        const QString filePath = it.next();
        ret.append(filePath);
    }
    return ret;
```

Reply:
@Chocobo1   
Are you sure it will compile? (Returning value in the void method)

Reply:
>Are you sure it will compile? (Returning value in the void method)

Oops, I fixed it.

Reply:
This code does not compile
```
[build] /workspace/qbittorrent/src/webui/api/appcontroller.cpp: In member function 'void AppController::getDirectoryContentAction()':
[build] /workspace/qbittorrent/src/webui/api/appcontroller.cpp:1114:18: error: variable 'QDirIterator it' has initializer but incomplete type
[build]  1114 |     QDirIterator it {dir_path, (QDir::Dirs | QDir::NoDotAndDotDot)};
[build]       |                  ^~
[build] ninja: build stopped: subcommand failed.
```

Moreover, it makes code much less readable.

> I would avoid redundant temporary string list

I don't think this is a problem. The compiler should take care of this variable.

Reply:
> This code does not compile

It's because you didn't include corresponding header (where `QDirIterator` is declared).

Reply:
🤦

Reply:
But I still find this code messy, less readable and not much better optimized.

Reply:
>But I still find this code messy, less readable and not much better optimized.

It is ok you don't do it. I can do a follow up commit later.

>The compiler should take care of this variable.

To be really sure, you'll need to look at the generated assembly. And I don't really think C++ compilers is capable of that.